tap_dynamodb/sync_strategies/full_table.py

```diff
@@ -101,3 +116,71 @@ def sync(config, state, stream):
     singer.write_version(table_name, stream_version)
 
     return rows_saved
+
+def get_expr_names(projection_element, expression_list):
+    '''
+    Prepare expression names for general expression attributes as well as for each of the nested
+    expression attributes.
+    For example:
+    projections = ["Comment", "Ticket"], expressions = ["Comment"]
+    return = "#Commnt"
+    Example of nested expression :
+    projections = ["Name[0].Comment"], expressions = ["Name.Comment"]
+    return = "#Namme[0]" and "#Commnt" in the second loop call
+    '''
+    half_length = int(len(projection_element)/2)
+    expr = "#{}".format(projection_element[:half_length+1]+projection_element[-2:]) # prepare expression attribute name
+    # remove `[` and `]` from the expression name
+    expr = expr.replace("[", "").replace("]", "")
+    # if the projection is a list, it will contain the index in `[]`, hence we need to split
+    # through '[' to get the index stored in the projection inside `[]` and append the remaining
+    # to the newly created `expr`.
+    proj_split = projection_element.split('[', 1)
+    expression_list[expr] = proj_split[0]
+    # if the length of the split is 1, it means there is no `[]` in the projection,
+    # hence there's no need to append the index. If the length is 1, the second element
+    # of the `proj_split` would be appended to the `expr`.
+    if len(proj_split) != 1:
+        expr = expr + '[' + proj_split[1]
+    return expr
+
+def prepare_expression(projections, expressions):
+    """
+    Prepare expression attributes for reserved word. Loop through all projections.
+    If projection found in expressions(reserved word list) then prepare expression attribute name for the same with
+    starting of # sign followed by the combination of 1st half part of projection, next character of 1st half and last 2 character of projection.
+    Because as per the documentation an expression attribute name must begin with a pound sign (#), and be followed
+    by one or more alphanumeric characters. Prepare expression Dict element with key as expression attribute name and
+    value as projection and replace projection with expression attribute name in projections(list of fields that need
+    to be fetched). Return expression dict and comma seprated string of projections.
+    Example :
+    projections = ["Comment", "Ticket"], expressions = ["Comment"]
+    return = "#Commnt, Ticket", {"#Commnt" : "Comment"}
+    Example of nested expression :
+    projections = ["Name[0].Comment"], expressions = ["Name.Comment"]
+    return = "#Namme[0].#Commnt", {"#Namme": "Name", "#Commnt": "Comment"}
+    """
+    i = 0
+    expression_list = {}
+    # Loop through all projection.
+    for projection_element in projections:
```

Comment:
Please update the code to get the index and value during the for loop itself.
Ex: `for index, projection_element in enumerate(projections):`

Reply:
Done